File: year2/mpp/java/MotorcycleContestHibernate/MotorcycleServer/src/main/java/networking/RpcConcurrentServer.java
package networking;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import services.MotorcycleService;

import java.net.Socket;

public class RpcConcurrentServer extends AbstractConcurrentServer {
    private static final Logger LOGGER = LogManager.getLogger(RpcConcurrentServer.class.getName());
    private MotorcycleService motorcycleService;

    public RpcConcurrentServer(int port, MotorcycleService service) {
        super(port);
        LOGGER.info("creating rpc server");
        this.motorcycleService = service;
    }

    public RpcConcurrentServer() {
    }

    public void setMotorcycleService(MotorcycleService motorcycleService) {
        this.motorcycleService = motorcycleService;
    }

    @Override
    protected Thread createWorker(Socket client) {
        LOGGER.info("creating worker for client " + client);
        MotorcycleClientRpcWorker worker = new MotorcycleClientRpcWorker(motorcycleService, client);
        Thread workerThread = new Thread(worker);
        LOGGER.traceExit();
        return workerThread;
    }
}

File: year2/mpp/c#/MotorcycleClientServer/MotorcycleServer/Networking/RpcConcurrentServer.cs
using System.Net.Sockets;
using System.Threading;
using MotorcycleCommon.Services;
using MotorcycleContest.Domain.Dtos;

namespace MotorcycleServer.Networking
{
    public class RpcConcurrentServer : AbstractConcurrentServer
    {
        private IMotorcycleService service;

        public RpcConcurrentServer(string host, int port, IMotorcycleService service) : base(host, port)
        {
            // Logger.Info("creating rpc concurrent server");
            this.service = service;
        }

        protected override Thread CreateWorker(TcpClient client)
        {
            // Logger.Info("creating worker for client " + client.ToString());
            MotorcycleClientRpcWorker worker = new MotorcycleClientRpcWorker(service, client);
            return new Thread(worker.Run);
        }
    }
}